src/demo/ui.jsx

```diff
@@ -756,6 +756,19 @@ class Pond extends React.Component {
       }
     }
 
+    const nextButtonText =
+      state.appMode === AppMode.FishLong ? 'Play Again' : 'Continue';
+    const nextButtonOnClick = () => {
+      if (state.appMode === AppMode.FishLong) {
+        resetTraining();
+        toMode(Modes.Words);
```

Comment:
Philosophical question around whether `toMode` should be responsible for `resetTraining` but no strong opinion.

Reply:
I don't think so because we also have "Train Again" where we definitely don't want to reset training.